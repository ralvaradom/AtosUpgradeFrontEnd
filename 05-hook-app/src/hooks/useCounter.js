import { useState } from "react";


const useCounter = (initialState=1, factor=1) => {
	const [ counter, setCounter ] = useState(initialState);

	const increment = () => {
		setCounter(counter+factor);
	}

	const decrement = () => {
		setCounter(counter-factor);	
	}

	const reset = () => {
		setCounter(initialState);
	}

	return {
		counter,
		increment,
		decrement,
		reset,
	};
};

export default useCounter;
