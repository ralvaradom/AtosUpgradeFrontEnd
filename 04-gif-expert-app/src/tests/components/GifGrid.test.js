import { shallow } from 'enzyme';
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

	test('debe mostararse correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});

		const wrapper = shallow(<GifGrid category='One punch' />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe mostrar items cuando se cargan las imagenes', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://localhost/cualquiercosa/cosa.jpg',
				title: 'cualquiercosa'
			},
			{
				id: 'ABC',
				url: 'https://localhost/cualquiercosa/cosa.jpg',
				title: 'cualquiercosa'
			},
		]

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category='One punch' />);
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});