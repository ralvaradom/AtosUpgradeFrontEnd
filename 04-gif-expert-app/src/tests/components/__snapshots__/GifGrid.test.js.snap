// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostararse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
